### PR TITLE
doc: release notes: add initial content for 2.1.0

### PR DESCRIPTION
**Pull request by dleach02**
Adding initial pass of content for v2.1.0 release preparation.

Signed-off-by: David Leach <david.leach@nxp.com>

### CODE REVIEW

File: doc/releases/release-notes-2.1.rst [hunk 1]
```diff
@@ -17,53 +17,304 @@ Kernel
...
+20475 : coccinelle: cleanups related to timeout : pabigot : Peter A. Bigot
+19650 : coccinelle: final pass standardizing integer timeout parameters : pabigot : Peter A. Bigot
+19623 : coccinelle: improve integer timeout script and fix missed cases : pabigot : Peter A. Bigot
+19562 : coccinelle: standardize kernel timeout arguments : pabigot : Peter A. Bigot
+
+
 Networking
 **********
 
-* Added new TCP stack implementation. The new TCP stack is still experimental
```
Comment (galak) [inline doc/releases/release-notes-2.1.rst]: You removed a bunch stuff..

**Review by galak — CHANGES_REQUESTED** [review]
Fix this to not remove what people have already put in.

File: doc/releases/release-notes-2.1.rst [hunk 2]
```diff
@@ -17,53 +17,304 @@ Kernel
 ******
 
 * TBD
+19807 : kernel: activate arch interface headers : andrewboie : Andrew Boie
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: Need a blank line after the last item in a bullet list.

File: doc/releases/release-notes-2.1.rst [hunk 3]
```diff
@@ -141,4 +660,305 @@ release:
...
+* :github:`17430` - arch/x86: drivers/interrupt_controller/system_apic.c improperly classifies IRQs
+* :github:`17361` - _THREAD_QUEUED overlaps with x86 _EXC_ACTIVE in k_thread.thread_state
+* :github:`17337` - ArmV7-M mpu sub region alignment
+* :github:`17239` - Too many open files crash when running "sanitycheck" with no arguments
+* :github:`17234` - CONFIG_KERNEL_ENTRY appears to be superfluous
+* :github:`17133` - arch/x86: x2APIC EOI should be inline
+* :github:`17104` - arch/x86: fix -march flag for Apollo Lake
+* :github:`17064` - drivers/serial/uart_ns16550: CMD_SET_DLF should be removed
+* :github:`16900` - Inline assembly in Arm z_arch_switch_to_main_thread missing clobber list
+* :github:`16880` - Systematic *-zephyr-eabi/bin/ld: warning: toolchain_is_ok cannot find entry symbol _start; defaulting to 000::00XXXXX
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: The script that creates these lists of issues needs to scan for meaningful reST syntax and escape those so they're not interpreted by Sphinx.  In this case replace ``*`` with ``\*``

File: doc/releases/release-notes-2.1.rst [hunk 4]
```diff
@@ -141,4 +660,305 @@ release:
...
+* :github:`12535` - Bluetooth: suspend private address (RPA) rotating
+* :github:`12509` - Fix rounding in _ms_to_ticks()
+* :github:`12504` - STM32: add USB_OTG_HS example
+* :github:`12206` - OpenThread apps want to download and build OpenThread every time!
+* :github:`12114` - assertion using nRF5 power clock with BLE and nRF5 temp sensor
+* :github:`11743` - logging: add user mode access
+* :github:`11717` - qemu_x86 's SeaBIOS clears the screen every time it runs
+* :github:`11655` - Alleged multiple design and implementation issues with logging
+* :github:`10748` - Work waiting on pollable objects
+* :github:`10701` - API: Prefix (aio_) conflict between POSIX AsyncIO and Designware AnalogIO Comparator
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: The script that creates these lists of issues needs to scan for meaningful reST syntax and escape those so they're not interpreted by Sphinx.  In this case replace ``_`` with ``\_``

File: doc/releases/release-notes-2.1.rst [hunk 5]
```diff
@@ -89,28 +340,162 @@ Bluetooth
...
 * TBD
 
+16874 : Bluetooth: Allow disconnected state connections to be reconnected : jhedberg : Johan Hedberg
+
+18338 : bluetooth: controller: Added function to support RX link quota update : mtpr-ot : Morten Priess
+18477 : bluetooth: controller: Add vendor operations to TICKER_USER_ULL_HIGH_OPS : mtpr-ot : Morten Priess
+20254 : bluetooth: controller: Add vendor specific overhead to TX buffer size : mtpr-ot : Morten Priess
+18327 : bluetooth: controller: Connection meta property support in LLL : mtpr-ot : Morten Priess
+19657 : Bluetooth: controller: Handle Read Remote version as a priority event : carlescufi : Carles Cufí
+19240 : Bluetooth: controller: legacy: Ensure that a `case` statement is present : cvinayak : Vinayak Kariappa Chettimada
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: The script that creates these lists of issues needs to scan for meaningful reST syntax and escape those so they're not interpreted by Sphinx.  In this case replace `` ` `` with ``\` ``  (or better would be to change `` ` `` (back-tick) to `` ' `` (apostrophe)

File: doc/releases/release-notes-2.1.rst [hunk 6]
```diff
@@ -141,4 +660,305 @@ release:
...
+* :github:`19638` - Bluetooth: Mesh: Provisioning Over PB-ADV
+* :github:`19629` - tinycbor buffer overflow causing mcumgr image upload failure
+* :github:`19612` - ICMPv6 packet is routed to wrong interface when peer is not found in neighbor cache
+* :github:`19604` - Bluetooth: ATT does not release all buffers on disconnect
+* :github:`19603` - addition to winbond,w25q16.yaml required for SPI CS to be controlled by driver
+* :github:`19592` - Request new repository to host the Eclipse plugin for building Zephyr applications
+* :github:`19569` - nRF RTC Counter with compile time decision about support of custom top value
+* :github:`19560` - Console on CDC USB  crashes when CONFIG_USB_COMPOSITE_DEVICE=y
+* :github:`19552` - [TOPIC-GPIO]: Support for legacy interrupt configuration breaks new API contract
+* :github:`19550` - drivers/pcie: `pcie_get_mbar()` should return a `void *` not `u32_t`
```
Comment (dbkinder) [inline doc/releases/release-notes-2.1.rst]: The script that creates these lists of issues needs to scan for meaningful reST syntax and escape those so they're not interpreted by Sphinx.  In this case replace `` ` `` with `` ' `` (back-tick with apostrophe)

**Review by dbkinder — CHANGES_REQUESTED** [review]
Need to update the script that creates the list of github issues that are included here to modify syntax that's meaningful to reST in the GitHub issue titles:  escaping ``\*`` and ``\_``, and changing back-tick `` ` `` to apostrophe `` ' ``

**Comment by dbkinder** [comment]
(alternatively, we can edit the offending issue titles in github)

File: doc/releases/release-notes-2.1.rst (hunk 1 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: yeah, I realized after that the content was not just temp filler.

File: doc/releases/release-notes-2.1.rst (hunk 4 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: ok

File: doc/releases/release-notes-2.1.rst (hunk 2 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: These were meant to be just highlight notes to the community to remind them of PRs that appear to belong to this heading. I didn't attempt to format them like we do with the issues closed section.

I'll clarify on next update

File: doc/releases/release-notes-2.1.rst (hunk 3 above)
Comment (dleach02) [reply doc/releases/release-notes-2.1.rst]: Okay. Will do.

**Comment by dleach02** [comment]
@dbkinder , is there a list of symbols someplace that I should reference to create escape sequences for reST syntax?

**Pull request closed without merging**